tests/testthat/test-as-data-frame.R

```diff
@@ -228,6 +224,25 @@ with_test_authentication({
         expect_true(is.data.frame(as.data.frame(ds, force=TRUE)))
     })
 
+    ds$hidden_var <- 1:20
+    ds <- hideVariables(ds, "hidden_var")
+
+    test_that("as.data.frame(force) retrieves hidden variables", {
+        skip_locally("Vagrant host doesn't serve files correctly")
+        expect_equal(hiddenVariables(ds), "hidden_var")
+
+        expect_warning(df <- as.data.frame(ds, force = TRUE),
+            "Variable hidden_var is hidden")
+        expect_equal(names(df), c("v1", "v2", "v3", "v4", "v5", "v6", "hidden_var"))
+
+        expect_warning(
+            df <- as.data.frame(ds[, c("v1", "hidden_var")], force = TRUE),
+            "Variable hidden_var is hidden"
+        )
+        expect_equal(names(df),
+            c("v1", "hidden_var"))
```

Comment:
This looks like the right behavior / expectation to me: If I explicitly ask for a variable (that happens to be hidden) I get that variable (along with any other variables I might have requested)